scene/resources/animation.h

```diff
@@ -77,6 +77,12 @@ class Animation : public Resource {
 		HANDLE_MODE_BALANCED,
 	};
 
+	enum RetargetMode {
```

Comment:
It does not seem correct that retarget information has to exist in Animation. What is the rationale behind this?

Reply:
@reduz It is obviously redundant to always have two Skeletons and need a SkeletonRetarget node when using retargeted animations for several models.

If you don't include that mode in your animation, we can store the Rest itself in the animation data, and for example, for rotation, you will need to multiply the quaternion 6 times during playback.

However, with the method I implemented, we can split the process into two parts: extract and apply. When playback animation, all that is required is 3 multiplications in the apply process.